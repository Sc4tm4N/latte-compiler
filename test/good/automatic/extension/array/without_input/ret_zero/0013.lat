int[] x() {
  return new int[3];
}

int[] y() {
  int[] x = x();
  x[1] = 5;
  return x;
}

int[] z(int[] z) {
  z[0] = 10;
  z[1] = 20;
  z[2] = 30;
  return z;
}

int main() {
  printInt(y()[1]);
  int[] x = y();
  printInt(x[0]);
  printInt(x[1]);
  printInt(x[2]);
  int[] g = z(x);
  printInt(g[0]);
  printInt(g[1]);
  printInt(g[2]);
  printInt(x[0]);
  printInt(x[1]);
  printInt(x[2]);
  g[0] = 100;
  g[1] = 200;
  g[2] = 300;
  printInt(g[0]);
  printInt(g[1]);
  printInt(g[2]);
  printInt(x[0]);
  printInt(x[1]);
  printInt(x[2]);
  return 0;
}