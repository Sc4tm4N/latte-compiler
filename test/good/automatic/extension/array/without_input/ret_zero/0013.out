5
0
5
0
10
20
30
10
20
30
100
200
300
100
200
300
